twarep/electrifly
# Finished updating the UI

## Checklist
- [N] Did I add this pull request into the appropriate Notion ticket? If not, create a ticket for this PR.

## What I Did
There are many things. Hold on to your hats
- Changed entire backend of `model_querying.py` to be able to query to the model table in the db and make predictions. It is also made safe as I hid the database connections behind hidden variables and columns that can only be called in the specific function.
- Changed entire layout for the ml ui on `app.py`. Added colors and accordions to make it look better.
- Made changes to how the data is input. Either one value for each attribute or a range that is randomly selected from.

## How I Did it?
- Just good old Python and almost 12 hours or coding. 

## How to Test?
There are multiple things to manually test.
1. You need to test if the activities and the specific inputs for those activities is showing up in the table when you press `Add Activity`.
2. Then you have to click on one of the rows on the table and see if that row is deleted when you click `delete row`.
3. Then you need to add multiple activities in the flight and make sure that the text stating soc remaining changes. It will also change after you pass 30%. I didn't have time to figure out how to keep the person from adding more, but the text should stop them.
4. Then you need to check/uncheck the toggle. This basically allows you to either input only one value for the data, or specify a range of values that are randomly chosen by the system.
5. Change the dates/times around and see if it works.

### Relevant Links
Any tutorials or other resources used to help with this task. You can also cite the documentation.
